### PR TITLE
DAOS-623 test: Check for use of DER_UNKNOWN in logs.

### PR DESCRIPTION
**Pull request by ashleypittman**
Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -6205,9 +6205,8 @@ obj_list_shards_get(struct obj_auxi_args *obj_auxi, unsigned int map_ver,
 		grp_idx, *shard, *shard_cnt);
 
 out:
-	D_DEBUG(DB_IO, DF_OID" list on shard %u leader %s: "DF_RC"\n",
-		DP_OID(obj->cob_md.omd_id), *shard,
-		obj_auxi->to_leader ? "yes" : "no", DP_RC(rc));
+	D_DEBUG(DB_IO, DF_OID " list on shard %u leader %s: %d\n", DP_OID(obj->cob_md.omd_id),
+		*shard, obj_auxi->to_leader ? "yes" : "no", rc);
```
Comment (daltonbohning) [inline src/object/cli_obj.c:6209]: why not DP_RC?

**Review by daltonbohning — APPROVED** [review]
Would this need to go into 2.4 as well? If so, we'll probably need a standalone ticket

**Comment by ashleypittman** [comment]
> Would this need to go into 2.4 as well? If so, we'll probably need a standalone ticket

I don't think so, this is just clearing the way for further checking of the logs.

> If so, we'll probably need a standalone ticket

Agree on this side.

File: src/object/cli_obj.c (hunk 1 above)
Comment (ashleypittman) [reply src/object/cli_obj.c:6209]: This is one of those functions where negative is a error code and positive has another meaning.

**Review by jolivier23 — APPROVED** [review]

**Comment by jolivier23** [comment]
My only concern is that I'd rather see DER_UNKNOWN(1) on occasion if it means I'll get a proper DER_ when it is an error.  Perhaps the way to handle such cases is to do something like 

If (D_LOG_ENABLED(DB_IO) {
   if (rc > 0)
       D_DEBUG(..., "%d", rc);
   else
      D_DEBUG(..., DF_RC, DP_RC(rc));
}

Perhaps we need a D_DEBUG_RC(DB_IO, rc, .....) essentially does the above?

**Comment by ashleypittman** [comment]
> My only concern is that I'd rather see DER_UNKNOWN(1) on occasion if it means I'll get a proper DER_ when it is an error. Perhaps the way to handle such cases is to do something like

We shouldn't be using code where negative is an error code and positive is a actual value.  Outside of things like iterators where we can only return one value then it's adding risk for little benefit.  We also don't typically use DF_RC for DEBUG messages.

**Pull request merged**